pkg/abstractions/endpoint/buffer.go

```diff
@@ -349,9 +353,16 @@ func (rb *RequestBuffer) handleHttpRequest(req request) {
 
 	resp, err := httpClient.Do(httpReq)
 	if err != nil {
-		req.ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
-			"error": "Internal server error",
-		})
+		var netErr net.Error
+		if errors.As(err, &netErr) && netErr.Timeout() {
```

Comment:
this second condition could be cleaned up

Reply:
cleaned up